### PR TITLE
Simulate/Reset Exposures

### PR DESCRIPTION
**Pull request by mattThousand**
### Why
We would like to show real exposure data from the `Exposure Notifications` API to the user.

### This Commit
This commit creates an event emitter that updates the JS layer when exposures are added or removed in the native iOS layer.

IMG_1841

Screen Shot 2020-06-16 at 1 49 23 PM

### CODE REVIEW

File: app/ExposureHistoryContext.tsx [hunk 1]
```diff
@@ -94,6 +95,25 @@ const ExposureHistoryProvider = ({
   const toggleHasExposure = () => {
     setHasBeenExposed(!hasBeenExposed);
   };
+  const eventEmitter = new NativeEventEmitter(
+    NativeModules.ExposureEventEmitter,
+  );
+
+  useEffect(() => {
+    NativeModules.ExposureEventEmitter.startListening();
```
Comment (johnschoeman) [inline app/ExposureHistoryContext.tsx]:
```suggestion
    const unsubscribe = NativeModules.ExposureEventEmitter.startListening();
    
    return unsubscribe
```

**Review by johnschoeman — APPROVED** [review]
:SHIPEITE:

File: app/exposureNotificationsNativeModule.ts [hunk 2]
```diff
@@ -125,20 +126,23 @@ export const resetExposures = async (
...
   cb: exposureEventCallback,
-): EventSubscription => {
-  return ExposureEvents.addListener(
-    'EXPOSURES_CHANGED',
-    (jsonString: string) => {
-      const exposures: Possible[] = JSON.parse(jsonString);
-      cb(exposures);
-    },
-  );
+): EventSubscription | null => {
```
Comment (mattThousand) [inline app/exposureNotificationsNativeModule.ts:133]: @johnschoeman this feels somewhat grotesque, but allocating `ExposureEvents` in the root scope caused an invariant error when running `GPS`. Would welcome an alternative suggestion, I'm not sure how idiomatic optionals are in `typescript`

Comment (johnschoeman) [reply app/exposureNotificationsNativeModule.ts:133]: If you have a Maybe in your domain, then you have a Maybe in your domain. The best practice of trying to reduce and move Maybes to the edges of your system still applies, but ts doesnt have super nice of ergonomics around Maybe types so you do see a lot of the `MyThing | null` syntax that you have here.

Comment (johnschoeman) [reply app/exposureNotificationsNativeModule.ts:133]: that said i would try to pull the conditional up as much as possible in the flow.

**Pull request merged**